cmake/environmentchecks.cmake

```diff
@@ -56,29 +70,39 @@ else()
 endif()
 
 if(CLANG_FORMAT_EXE AND GIT_EXE)
-  set(CLANG_FORMAT_TESTING_AVAILABLE TRUE CACHE BOOL "Executables required to run the ClangFormat test are available.")
+  set(CLANG_FORMAT_TESTING_AVAILABLE TRUE INTERNAL BOOL "Executables required to run the ClangFormat test are available.")
 endif()
 
 if(CLANG_TIDY_EXE)
-  set(CLANG_TIDY_AVAILABLE TRUE CACHE BOOL "Executable required to perform static analysis is available.")
+  set(CLANG_TIDY_AVAILABLE TRUE INTERNAL BOOL "Executable required to perform static analysis is available.")
 endif()
 
-if(FIND_EXE AND GCOV_EXE AND Python_Interpreter_FOUND)
-  set(COVERAGE_TESTING_AVAILABLE TRUE CACHE BOOL "Executables required to run the coverage testing are available.")
+if(FIND_EXE AND GCOV_EXE AND Python_Interpreter_FOUND AND GCC_COVERAGE_COMPILER_FLAGS_OK)
+  set(COVERAGE_TESTING_AVAILABLE TRUE INTERNAL BOOL "Executables required to run the coverage testing are available.")
 endif()
 
 if(SWIG_EXECUTABLE)
-  set(BINDINGS_AVAILABLE TRUE CACHE BOOL "Executable required to generate bindings is available.")
+  set(BINDINGS_AVAILABLE TRUE INTERNAL BOOL "Executable required to generate bindings is available.")
 endif()
 
 if(VALGRIND_EXE AND Python_Interpreter_FOUND)
-  set(VALGRIND_TESTING_AVAILABLE TRUE CACHE BOOL "Executable required to run valgrind testing is available.")
+  set(VALGRIND_TESTING_AVAILABLE TRUE INTERNAL BOOL "Executable required to run valgrind testing is available.")
+endif()
+
+if(LLVM_PROFDATA_EXE AND LLVM_COV_EXE AND FIND_EXE AND LLVM_COVERAGE_COMPILER_FLAGS_OK)
+  set(LLVM_COVERAGE_TESTING_AVAILABLE TRUE INTERNAL BOOL "Executables required to run the llvm coverage testing are available.")
 endif()
 
 mark_as_advanced(
-  CLANG_TIDY_EXE CLANG_TIDY_AVAILABLE
-  CLANG_FORMAT_EXE GIT_EXE CLANG_FORMAT_TESTING_AVAILABLE
-  FIND_EXE GCOV_EXE COVERAGE_TESTING_AVAILABLE
-  SWIG_EXECUTABLE BINDINGS_AVAILABLE
-  VALGRIND_EXE VALGRIND_TESTING_AVAILABLE
+  CLANG_TIDY_EXE
+  CLANG_FORMAT_EXE
+  GIT_EXE
+  FIND_EXE
+  GCOV_EXE
+  SWIG_EXECUTABLE
+  VALGRIND_EXE
+  LLVM_COV_EXE
+  LLVM_PROFDATA_EXE
+  LLVM_COVERAGE_COMPILER_FLAGS_OK
+  GCC_COVERAGE_COMPILER_FLAGS_OK
```

Comment:
Might want to keep that list alphabetically sorted?

Reply:
Ok